505 feature/tooltip on cron schedules

mariadanieldeepak opened the pull request:
Closes https://github.com/sudar/bulk-delete/issues/505

sudar commented:
@mariadanieldeepak 

I am seeing lot of unnecessary changes. Also looks like you have moved the function to a different class.

Did you branched off from the latest `dev/6.0.0` branch before start working on this?

mariadanieldeepak commented:
@sudar,

> Did you branched off from the latest dev/6.0.0 branch before start working on this?

Yes. I figured out what went wrong. I branched off w/o pulling the latest `dev/6.0.0`. I've fixed this and it should all be fine now.

Also, the issues shown above in Style CI is not because of the files I modified in this PR. I guess we'll have to deal that as a separate issue.

sudar commented:
@mariadanieldeepak 

Merge latest `dev/6.0.0` to your branch and it should take care of the Style CI issues.

Also show the help message only when the scheduler add-on is installed.

mariadanieldeepak commented:
@sudar,

I've modified the code to fix the architectural issues that you identified. Screenshot below portrays that the helper text is displayed when Scheduler add-on is activated and hides when it is not.

![image](https://user-images.githubusercontent.com/17510033/45818365-264c0a00-bcff-11e8-866c-021fd94a14b5.png)

However, not all modules respond to this code change. Let me give you an example.

I've the `Bulk Delete - Scheduler for Deleting Posts by Status Addon` activated (refer screenshot below)

![image](https://user-images.githubusercontent.com/17510033/45818485-804ccf80-bcff-11e8-98be-76ea2ff6c4d2.png)

But the helper text doesn't appear in the module because the add-on isn't listed under `pro_iterators` variable (`BulkWP.pro_iterators`). And that is because `BD_Scheduler_For_Deleting_Posts_By_Status` class isn't extending `BD_Scheduler_Addon` which adds the add-on slug using `filter_js_array()`

Screenshot: The helper text should be visible, but it is not.

![image](https://user-images.githubusercontent.com/17510033/45818555-aecaaa80-bcff-11e8-94c0-cc78cffa71b0.png)

Screenshot: `BD_Scheduler_For_Deleting_Posts_By_Status` is not including the slug in `pro_iterators`.

![image](https://user-images.githubusercontent.com/17510033/45818522-9a86ad80-bcff-11e8-88e1-1ac3fcceb7cc.png)

Do you want me to modify the Add-on's? (May be split this issue further?)

mariadanieldeepak commented on include/Core/Base/Mixin/Renderer.php:
```diff
@@ -408,22 +410,21 @@ protected function render_cron_settings() {
+				 */
+				apply_filters( 'bd_pro_only_feature_class', $pro_class, $this->field_slug, $this )
+				?>
 				<?php if ( ! empty( $this->scheduler_url ) ) : ?>
```
Swapping dev between ES6 scoping and PHP's ;) I've reset the changes now.

sudar commented on include/Core/Base/Mixin/Renderer.php:
```diff
@@ -432,9 +432,16 @@ protected function render_cron_settings() {
 		</tr>
 
-		<tr>
+		<tr class="<?php if ( ! empty( $pro_class ) ) { echo sanitize_html_class( $pro_class ); } ?>" style="display: none;">
```
You can remove the `empty` check and just print it after passing through `sanitize_html_class`.

mariadanieldeepak replied:
> you can remove the `empty` check and just print it after passing through `sanitize_html_class`.

Just realized that `sanitize_html_class` handles empty string. Removed the explicit check.

sudar commented on include/Core/Metas/Modules/DeleteUserMetaModule.php:
```diff
@@ -120,8 +120,15 @@ public function render() {
 				</tr>
 
 				<tr>
```
You may have to add the pro class here also right?

sudar commented:
@mariadanieldeepak 

Create a seperate issue for the add-ons.

mariadanieldeepak commented:
@sudar,

> Create a seperate issue for the add-ons.

- https://github.com/sudar/bulk-delete-scheduler-for-deleting-posts-by-status/issues/3
